src/Network/Wai/SAML2/Validation.hs

```diff
@@ -190,8 +191,8 @@ validateResponse cfg responseData = runExceptT $ do
     then throwError NotValid
     else pure ()
 
-    -- all checks out, return the assertion
-    pure assertion
+    -- all checks out, return the assertion and InResponseTo field
+    pure (assertion, responseInResponseTo samlResponse)
```

Comment:
I am wondering if it makes sense to single out this particular value from the response and return it. It seems plausible that we might want to return other fields from the response in the future, which would then require further changes to the API here. Perhaps we can just future-proof this by returning the entire response? 

Alternatively (or at the same time), perhaps it would make sense to introduce a new type which can store all the results from this function, so that it would be easier to add a new field there if it becomes necessary down the line.

Reply:
I actually agree, for some reason I thought the function was internal so this wouldn't matter, but it's exported.
I think it would be better just to return the entire `Response`

Reply:
Looking into this,
* https://github.com/mbg/wai-saml2/pull/31 splits `validateResponse` into `decodeResponse` and `validateSAMLResponse`.
* `decodeResponse` already returns the whole `Response`, so this issue is already solved

I'm not sure how best to proceed here. I could change `validateResponse` to return the `Response` as well, but `validateResponse` seems a bit superfluous to begin with, all it does is call `decodeResponse` and then `validateSAMLResponse`. 

I would suggest that we: 
* remove `validateResponse` 
* and rename `validateSAMLResponse` to `validateResponse`

Reply:
I am a bit confused by your suggestion. `validateResponse` is the main entry point to the library for common use. Why would we remove it? The change in #31 is just there to allow parts of `validateResponse` to be tested/used independently.

Reply:
OK, then I'll just change `validateResponse` to also return the entire `Response`